lib/Facebook.js

```diff
@@ -815,10 +816,274 @@ function Facebookbot(configuration) {
+
+            var body = {
+                'user' : user
+            };
+
+            request({
+                method: 'DELETE',
+                json: true,
+                body: body,
+                uri: uri
+            }, function(err, res, body) {
+                if (err) {
+                    facebook_botkit.log('Could not remove a user from a label');
+                    cb(err);
+                } else {
+                    if (body.error) {
+                        facebook_botkit.log('ERROR in removing a user from a label API call : ', body.error);
+                        cb(body.error);
+                    } else {
+                        facebook_botkit.debug('Successfully remove a user from a label', body);
+                        cb(null, body);
+                    }
+                }
+            });
+        },
+        get_labels_by_user: function(user, cb) {
+            var uri = 'https://' + api_host + '/' + api_version + '/' + user + '/custom_labels?access_token=' + configuration.access_token;
 
+            request({
+                method: 'GET',
+                json: true,
+                uri: uri
+            }, function(err, res, body) {
+                if (err) {
+                    facebook_botkit.log('Could not get user associated labels');
+                    cb(err);
+                } else {
+                    if (body.error) {
+                        facebook_botkit.log('ERROR in getting user associated labels API call : ', body.error.message);
+                        cb(body.error);
+                    } else {
+                        facebook_botkit.log('Successfully get user associated labels');
+                        cb(null, body);
+                    }
+                }
+            });
+        },
+        get_label_details: function(label_id, fields, cb) {
+            var uri = 'https://' + api_host + '/' + api_version + '/' + label_id + '?access_token=' + configuration.access_token;
 
+            if (fields && fields.length > 0) {
+               uri += '&fields=' + fields.join(",");
+            }
+
+            request({
+                method: 'GET',
+                json: true,
+                uri: uri
+            }, function(err, res, body) {
+                if (err) {
+                    facebook_botkit.log('Could not get label details');
+                    cb(err);
+                } else {
+                    if (body.error) {
+                        facebook_botkit.log('ERROR in getting label details call : ', body.error.message);
+                        cb(body.error);
+                    } else {
+                        facebook_botkit.log('Successfully get label details');
+                        cb(null, body);
+                    }
+                }
+            });
+        },
+        get_all_label: function(fields, cb) {
```

Comment:
Maybe get_all_label should be get_all_labels with an s? there is get_labels_by_user above

Same comment about the fields, I think we can just grab all fields by default since FB only lets us specify `name` as an additional field currently

Reply:
Nice catch 👍